src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -145,6 +165,13 @@ def copy_files_and_dirs(src, dst):
         shutil.copy(src, dst)
 
 
+def remove_file(file_path: str):
```

Comment:
NIT: There is only one reference to this helper function, I think we could just inline this try catch block.